platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventValidationChecks.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.validation;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.INVALID_EVENT_ERROR;
+import static com.swirlds.platform.consensus.GraphGenerations.FIRST_GENERATION;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.metrics.LongAccumulator;
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.system.events.BaseEventHashedData;
+import com.swirlds.common.utility.CommonUtils;
+import com.swirlds.common.utility.throttle.RateLimitedLogger;
+import com.swirlds.platform.crypto.SignatureVerifier;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.security.PublicKey;
+
+/**
+ * A collection of static methods for validating events
+ */
+public class EventValidationChecks {
+    /**
+     * Hidden constructor
+     */
+    private EventValidationChecks() {}
+
+    /**
+     * Determine whether a given event has a valid generation.
+     * <p>
+     * This validation method doesn't include metrics, since ancient events are shed throughout the intake process, not
+     * just during validation.
+     *
+     * @param event                       the event to be validated
+     * @param minimumGenerationNonAncient the minimum generation for non-ancient events
+     * @param logger                      a logger for validation errors
+     * @return true if the event has a valid generation, otherwise false
+     */
+    public static boolean isGenerationValid(
+            @NonNull final GossipEvent event,
+            final long minimumGenerationNonAncient,
+            @NonNull final RateLimitedLogger logger) {
+
+        final boolean generationValid = event.getGeneration() >= minimumGenerationNonAncient;
+        if (!generationValid) {
+            logger.error(
+                    INVALID_EVENT_ERROR.getMarker(),
+                    "Event generation is invalid. Event: {}, Minimum Generation non ancient: {}",
+                    event,
+                    minimumGenerationNonAncient);
+        }
```

Comment:
I'm not sure this should be logged. In the normal course of business we are going to throw out ancient events on a regular basis any time a node falls behind.